src/FixedDataTableCellGroup.js

```diff
@@ -76,25 +87,70 @@ var FixedDataTableCellGroupImpl = React.createClass({
     }, false);
 
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
+    var count = 0;
+
+    var newColumnsToRender = new Array(props.maxVisibleColumns);
+    var positions = new Array(props.maxVisibleColumns);
+
+    for (var i = 0; i < columns.length; i++) {
       var columnProps = columns[i].props;
       var recycable = columnProps.allowCellsRecycling && !isColumnReordering;
       if (!recycable || (
-            currentPosition - props.left <= props.width &&
-            currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
+          currentPosition - props.left <= props.width &&
+          currentPosition - props.left + columnProps.width >= 0)) {
+        positions[i] = currentPosition;
```

Comment:
A couple things seem off to me here.
1) i is an index from 0 to columns.length, not 0 to maxVisibleColumns... should we be using count instead?
2) This if statement is called not only for the columns rendered on screen, but also any columns which are not recyclable.  It doesn't seem to take that into account.  Do we need to split out the onscreen part of the if statement to address that?